addon/components/frost-link.js

```diff
@@ -243,11 +243,46 @@ export default LinkComponent.extend(SpreadMixin, PropTypeMixin, {
 
   // == Lifecycle Hooks =======================================================
 
-  /* Ember.Component method */
+  /**
+   * Explicitly overrides the didReceiveAttrs in link-to
+   * https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-glimmer/lib/components/link-to.js#L790
+   *
+   * This allows named parameters to be passed to the link instead of positional
+   * params.  However, because the link-to expects positional params the named
+   * parameters must be set back against the component as a positional param
+   * array in the correct order (text, route, queryParams, models).  Once set
+   * we're free to hand control back to the parent function and it will react
+   * as if we used the original link-to interface.
+   */
+  didReceiveAttrs () {
+    if (isEmpty(this.get('params'))) {
+      let params = []
+
+      params.push(this.get('text'))
+
+      params.push(this.get('route'))
+
+      const queryParams = this.get('queryParams')
+      if (isPresent(queryParams)) {
+        params.push(queryParams)
+      }
+
+      const models = this.get('models')
+      if (!isEmpty(models)) {
+        params.push(this.get('models'))
```

Comment:
We should expect models to be an array instead of an object, so probably do a forEach/push here? Models should also come before queryParams (text, route, models, queryParams)https://github.com/emberjs/ember.js/blob/v2.8.0/packages/ember-htmlbars/lib/components/link-to.js#L805